common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -140,6 +146,79 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+        final String methodTag = TAG + ":onResume";
+        final Bundle extras = getExtras();
+        if (isSwitchBrowserResumeFlow(extras)) {
+            Logger.info(methodTag, "Resume switch browser protocol on WebView.");
+            resumeSwitchBrowser(extras);
+        }
+    }
+
+    /**
+     * Get the extras from the activity intent.
+     *
+     * @return Bundle with the extras
+     */
+    @NonNull
+    private Bundle getExtras() {
+        final Activity activity = getActivity();
+        if (activity == null) {
+            return Bundle.EMPTY;
+        }
+        final Intent intent = activity.getIntent();
+        if (intent == null) {
+            return Bundle.EMPTY;
+        }
+        final Bundle extras =  intent.getExtras();
+        return extras == null ? Bundle.EMPTY : extras;
+    }
+
+    /**
+     * Check if the extras contains the switch browser code and action uri.
+     * if so, it means we are resuming the switch browser flow.
+     *
+     * @param extras Bundle
+     * @return boolean
+     */
+    private boolean isSwitchBrowserResumeFlow(@NonNull final Bundle extras) {
+        final boolean containsActionUri = extras.containsKey(
+                AuthenticationConstants.SWITCH_BROWSER.ACTION_URI
+        );
+        final boolean containsCode = extras.containsKey(
+                AuthenticationConstants.SWITCH_BROWSER.CODE
+        );
+        return containsCode && containsActionUri;
```

Comment:
Should we also check if the existing state of the authorization is it is waiting for resuming switch browser flow? You may want store some state what you can validate here.

Reply:
Done [6f82f9d](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2557/commits/6f82f9da74f33171a8e39032585542dd09eaa4d9)